stdlib/configparser.pyi

```diff
@@ -29,6 +29,9 @@ __all__ = [
     "MAX_INTERPOLATION_DEPTH",
 ]
 
+if sys.version_info >= (3, 12):
+    __all__.remove("SafeConfigParser")
```

Comment:
Is that supported? I think we should instead .append it if we're below 3.12.

Reply:
I thought it was, but can't immediately find the reference, and I guess it's better to be safe than sorry :)

Reply:
At least typeshed-client only supports `+=`, and that's the only operation used in any existing stub in typeshed.

Reply:
Looks like pyright definitely supports it: https://github.com/microsoft/pyright/blob/main/docs/typed-libraries.md#library-interface

But whereas I can see code in `mypy/semanal.py` that specifically deals with `__all__.append` and `__all__.extend`...
https://github.com/python/mypy/blob/6b1fc865902bf2b845d3c58b6b9973b5a412241f/mypy/semanal.py#L4881-L4895

...I can't find any equivalent handling of `__all__.remove` in mypy.

Reply:
We should probably add a lint to flake8-pyi so that it forbids operations on `__all__` unless they're known to be supported by all major type checkers.

Reply:
Looking at the mypy code they also support `.append` and `.extend` but not `.remove`.

Reply:
Opened:

- https://github.com/PyCQA/flake8-pyi/issues/369